### PR TITLE
CASM-3727: Write out product catalog data

### PR DESCRIPTION
**Pull request by jsollom-hpe**
## Summary and Scope

Write out product catalog data to the output file specified by ${SCRIPT_OUTPUT_FILE}.
Made some other minor tweaks defining variables once rather than running basename repeatedly.

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves CASM-3727

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * `Frigg`

### Test description:

I ran a workflow where I uploaded a Loftsman manifest and made sure that manifest was recorded in the product catalog under the correct product.

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why? No
- Was upgrade tested? If not, why? No
- Was downgrade tested? If not, why? No
- Were new tests (or test issues/Jiras) created for this change? No, this is new code.

## Risks and Mitigations
Low.

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: bin/loftsman-manifest-upload [hunk 1]
```diff
@@ -80,11 +82,15 @@ process(){
                 fi
                 rm -f $name
             fi
-            if ! cray artifacts create config-data argo/loftsman/$PRODUCT_NAME/$PRODUCT_VERSION/manifests/"$(basename $manifest)" $manifest; then
-                echo "ERROR failed to create s3://config-data/argo/loftsman/$PRODUCT_NAME/$PRODUCT_VERSION/manifests/$(basename $manifest)"
+            manifest_name="$(basename $manifest)"
+            s3_manifest_key="argo/loftsman/$PRODUCT_NAME/$PRODUCT_VERSION/manifests/$manifest_name"
+            s3_manifest_path="config-data/$s3_anifest_key"
```
Comment (heemstra) [inline bin/loftsman-manifest-upload]: I assume s3_anifest should be s3_manifest here...

**Review by heemstra — CHANGES_REQUESTED** [review]

File: .github/workflows/iuf-container.yaml [hunk 2]
```diff
@@ -34,14 +34,14 @@ jobs:
       contents: read
       id-token: write
     env:
-      DOCKER_REPO: artifactory.algol60.net/csm-docker/stable/iuf
-      DOCKER_TAG: v0.1.1
+      DOCKER_REPO: artifactory.algol60.net/csm-docker/unstable/iuf
```
Comment (ekamin-hpe) [inline .github/workflows/iuf-container.yaml]: I'm guessing the changes in this file were left over from testing and will be removed before you merge.

**Review by ekamin-hpe — APPROVED** [review]
Looks good. Just for the record it appears to me that this does correctly handle the case when no loftsman manifests exist.

Edit: well, it does with 9a15b91 anyway.

File: bin/loftsman-manifest-upload (hunk 1 above)
Comment (jsollom-hpe) [reply bin/loftsman-manifest-upload]: Fixed.

File: .github/workflows/iuf-container.yaml (hunk 2 above)
Comment (jsollom-hpe) [reply .github/workflows/iuf-container.yaml]: Reverted.

**Comment by jsollom-hpe** [comment]
Talk with @bo-quan about whether there is actually a separate repository containing this Helm chart or not. The chart still appears in this repository.

**Review by heemstra — APPROVED** [review]

**Pull request merged**